dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ExecutionModel.java

```diff
@@ -308,12 +308,11 @@ private void extractReadsFrom() {
         	BigInteger address = addressedReads.getKey();
             for (EventData read : addressedReads.getValue()) {
                 for (EventData write : addressWritesMap.get(address)) {
-                    BoolExpr rfExpr = rf.getSMTVar(write.getEvent(), read.getEvent(), context);
-                    Expr rfInterp = model.getConstInterp(rfExpr);
+                    BooleanFormula rfExpr = rf.getSMTVar(write.getEvent(), read.getEvent(), context);
                     // The null check is important: Currently there are cases where no rf-edge between
                     // init writes and loads get encoded (in case of arrays/structs). This is usually no problem,
                     // since in a well-initialized program, the init write should not be readable anyway.
-                    if (rfInterp != null && rfInterp.isTrue()) {
+                    if (model.evaluate(rfExpr) != null && model.evaluate(rfExpr)) {
```

Comment:
Call model.evaluate only once (just like the old code did). Something like this:
```
rfEval = model.evaluate(rfExpr)
if (rfEval != null && rfEval) {
    //...
}
```.

Reply:
What should be the behaviour if `rfEval == null`? In the old code the body iff the if would just not be executed. If this is the intended behaviour we can just do `if (rfEval)` right?

Reply:
As the code comment suggests, there are some cases where some rf edge is not encoded and in that case we generally assume that that read-from relation simply didn't happen (i.e. rf(e1, e2) not encoded == rf(e1, e2) is false). This can happen due to active set reasoning. ``` if (rfEval) ``` doesn't work because it implicitly unboxes, causing a NullPointerException if the edge was not encoded (rfEval is a boxed boolean)

Reply:
ok ... `if(null)` would throw and exception instead of consider that the condition is not satisfied.